src/Collie/Core.idr

```diff
@@ -53,15 +50,38 @@ data Any : (p : Command -> Type) -> (cmd : Command) -> Type where
     (parsedSub : Any p (field pos)) ->
     Any p cmd
 
+namespace Any
+
+  public export
+  map : {cmd : Command} -> ((cmd : Command) -> p cmd -> q cmd) ->
+    Any p cmd -> Any q cmd
+  map f (Here pcmd) = Here (f _ pcmd)
+  map f (There pos p) = There pos (map f p)
+
 public export
-record ParsedCommand (cmd : Command) where
+record ParsedCommand (f, g : Type -> Type) (cmd : Command) where
```

Comment:
The extra flexibility is great. I have a slight worry what these abstractions represent. 
Maybe we ought to make sure to have type-synonyms in user-facing types.